lib/internal/Magento/Framework/App/Utility/Files.php

```diff
@@ -1041,25 +1039,26 @@ protected function _accumulateFilesByPatterns(array $patterns, $filePattern, arr
     }
 
     /**
-     * Parse meta-info of a static file in module
+     * Search static files from all modules by the specified pattern and accumulate meta-info
      *
-     * @param string $file
-     * @return array
+     * @param string $area
+     * @param string $filePattern
+     * @param array $result
+     * @return void
      */
-    protected function _parseModuleStatic($file)
+    protected function _accumulateStaticFiles($area, $filePattern, array &$result)
     {
-        foreach ($this->componentRegistrar->getPaths(ComponentRegistrar::MODULE) as $moduleName => $modulePath) {
-            if (preg_match(
-                '/^' . preg_quote("{$modulePath}/", '/') . 'view\/([a-z]+)\/web\/(.+)$/i',
-                $file,
-                $matches
-            ) === 1
-            ) {
-                list(, $area, $filePath) = $matches;
-                return [$area, '', '', $moduleName, $filePath, $file];
+        foreach ($this->componentRegistrar->getPaths(ComponentRegistrar::MODULE) as $moduleName => $moduleDir) {
+            $moduleWebPath = $moduleDir . "/view/{$area}/web";
+
+            foreach (self::getFiles([$moduleWebPath], $filePattern) as $absolutePath) {
+                $localPath = substr($absolutePath, strlen($moduleDir) + 1);
+                if (preg_match('/^view\/([a-z]+)\/web\/(.+)$/i', $localPath, $matches) === 1) {
```

Comment:
Why do we need this matching at all? I'm not sure I get this logic

Reply:
As I understood this is needed to fetch area and path after /web/
Initially files are fetched for all areas